File: android/Detonator/src/main/java/com/iconshot/detonator/request/TextAreaBlurRequest.java
package com.iconshot.detonator.request;

import com.iconshot.detonator.Detonator;
import com.iconshot.detonator.element.TextAreaElement;
import com.iconshot.detonator.renderer.Edge;

public class TextAreaBlurRequest extends Request {
    public TextAreaBlurRequest(Detonator detonator, IncomingRequest incomingRequest) {
        super(detonator, incomingRequest);
    }

    public void run() {
        Edge edge = getComponentEdge();

        TextAreaElement element = (TextAreaElement) edge.children.get(0).element;

        element.blur(() -> end());
    }
}

File: ios/Sources/Detonator/Request/TextAreaBlurRequest.swift
import UIKit

class TextAreaBlurRequest: Request {
    override public func run() {
        let edge = getComponentEdge()!
        
        let elementEdge = edge.children[0]
        
        let view = elementEdge.element!.view as! TextAreaView
        
        view.resignFirstResponder()
        
        end()
    }
}
